interop/klr/fromNKI.hpp

```diff
@@ -0,0 +1,511 @@
+#pragma once
+
+#include "klir_ast.hpp"
+#include "klir_common.hpp"
+#include <variant>
+#include <string>
+#include <vector>
+#include <optional>
+
+namespace Trace {
+
+struct Error {
+    std::string msg;
+    Error(std::string&& m) : msg(std::move(m)) {}
+};
+
+// Simple error handling
+template<typename T>
+using TraceResult = std::variant<T, Error>;
+
+template<typename T>
+bool is_ok(const TraceResult<T>& result) {
+    return std::holds_alternative<T>(result);
+}
+
+template<typename T>
+T get_ok(const TraceResult<T>& result) {
+    return std::get<T>(result);
```

Comment:
This is returning a copy of T. Because it takes result by reference, but return a T by value.

We might want it to be more like [std::get for variant](https://en.cppreference.com/w/cpp/utility/variant/get.html) and [std::expected::value()](https://en.cppreference.com/w/cpp/utility/expected/value.html). Returning either a reference, or rvalue, to what's inside

```
template<typename T>
T& get_ok(const TraceResult<T>& result) {
    return std::get<T>(result);
}

template<typename T>
T&& get_ok(TraceResult<T>&& result) {
    return std::get<T>(std::move(result));
}
```